### PR TITLE
All claims 41 effective end date mg

### PR DESCRIPTION
**Pull request by markgreenburg**
## Description
Adds end date for effecitve dates to the All Claims form
Not sure if this content looks right.

## Testing done
- [x] local testing
- [x] unit tests

## Screenshots
When no end date:
screen shot 2018-10-10 at 10 29 01 pm

-----

When end date:
screen shot 2018-10-10 at 10 28 16 pm

## Acceptance criteria
- [x] Forwarding address has an end date
- [x] End date is optional

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Comment by cvalarida** [comment]
First blush: From the connected ticket, two of the requirements were:
- Effective date cannot be in the past (e.g. `04/05/1980`)
- End date must be in the future (e.g. not `04/02/2000`)

Judging by your screenshots, it looks like you still need some validation. :wink: 

But now I'll actually look at the code...

File: src/applications/disability-benefits/526EZ/helpers.jsx [hunk 1]
```diff
@@ -797,3 +798,20 @@ export const validateIfHasEvidence = (
 
 export const title10DatesRequired = formData =>
   get('view:isTitle10Activated', formData, false);
+
+export const ForwardingAddressViewField = ({ formData }) => {
+  const { effectiveDate } = formData;
+  return (
+    <div>
+      <EffectiveDateViewField formData={effectiveDate} />
+      <AddressViewField formData={formData} />
```
Comment (cvalarida) [inline src/applications/disability-benefits/526EZ/helpers.jsx]: @goldenmeanie Should the address in the screenshots be displayed differently? There's something that seems not quite right about them, but I can't put my finger on it. Maybe indent the address?

File: src/applications/disability-benefits/all-claims/tests/pages/contactInformation.unit.spec.jsx [hunk 2]
```diff
@@ -240,7 +240,9 @@ describe('Disability benefits 526EZ contact information', () => {
           },
           'view:hasForwardingAddress': true,
           forwardingAddress: {
-            effectiveDate: '2019-01-01',
+            effectiveDate: {
+              from: '2019-01-01',
```
Comment (cvalarida) [inline src/applications/disability-benefits/all-claims/tests/pages/contactInformation.unit.spec.jsx]: Should there be a test written for `effectiveDate.to`? Maybe that it displays as expected? :man_shrugging:

**Review by cvalarida — COMMENTED** [review]
Like I mentioned above, I think it just needs those validations. Otherwise, it looks great.

**Review by peggygannon — CHANGES_REQUESTED** [review]
@markgreenburg  Hi, there.. Just a few tweaks from me: 

- Please make We will >> We'll 
We'll use this address 

- Can we make the date in this format >> April, 5, 2018 rather than numerals? This would follow our style across the site. 
This has been our style for months: Jan., Feb., March, April, May, June, July, Aug., Sept., Oct., Nov., Dec.

File: src/applications/disability-benefits/526EZ/helpers.jsx (hunk 1 above)
Comment (goldenmeanie) [reply src/applications/disability-benefits/526EZ/helpers.jsx]: It looks like the lines are separate paragraphs instead of just linebreaks

Comment (goldenmeanie) [reply src/applications/disability-benefits/526EZ/helpers.jsx]: We do also have an indent-and-highlight address style that is inconsistently applied ;)

Comment (markgreenburg) [reply src/applications/disability-benefits/526EZ/helpers.jsx]: Yeah, this is much better. Could probably be its own PR, but I already did it, soo...
screen shot 2018-10-12 at 12 36 06 pm

File: src/applications/disability-benefits/all-claims/tests/pages/contactInformation.unit.spec.jsx (hunk 2 above)
Comment (markgreenburg) [reply src/applications/disability-benefits/all-claims/tests/pages/contactInformation.unit.spec.jsx]: Added some tests around validating the effective dates. And now it uses the common `dateRangeUI` uiSchema so I'm going to assume that's tested elsewhere.

**Comment by markgreenburg** [comment]
Thanks @peggygannon - here are some new screenshots for you:

'We’ll' instead of 'we will':
screen shot 2018-10-12 at 1 51 51 pm

screen shot 2018-10-12 at 1 51 42 pm

Consistent date formatting:
screen shot 2018-10-12 at 1 51 22 pm

screen shot 2018-10-12 at 1 51 42 pm

**Review by peggygannon — APPROVED** [review]

**Comment by markgreenburg** [comment]
@cvalarida could you please give this one another shot? Effective dates are now required to start in the future. The switch to the date range widget should take care of ensuring that the end date is after the start date.

File: src/applications/disability-benefits/all-claims/config/schema.js [hunk 3]
```diff
@@ -770,12 +781,13 @@ const schema = {
     // Forwarding address differs from mailing address in a few key ways:
     // 1. Address lines 1-3 are max 35 chars instead of 20
     // 2. The UI is such that requiring fields must be done in the UI schema
-    // 3. There is an effectiveDate property that specifies the date at which
-    //    the forwarding address should start to be used
+    // 3. There is are effectiveStartDate and effectiveEndDate properties that
```
Comment (cvalarida) [inline src/applications/disability-benefits/all-claims/config/schema.js]: Typo: "There is are"

...edito? :man_shrugging:

File: src/applications/disability-benefits/all-claims/content/contactInformation.jsx [hunk 4]
```diff
@@ -63,6 +58,20 @@ const EmailViewField = ({ formData, name }) => (
   </p>
 );
 
+const EffectiveDateViewField = ({ formData }) => {
+  const { from, to } = formData;
+  const dateFormat = 'MMM D, YYYY';
+  const fromDateString = moment(from).format(dateFormat);
+  return to ? (
+    <p>
+      We’ll use this address starting on {`${fromDateString} `}
```
Comment (cvalarida) [inline src/applications/disability-benefits/all-claims/content/contactInformation.jsx]: I feel like this is another case where Prettier makes not-so-pretty formatting decisions. :man_shrugging:

**Review by cvalarida — APPROVED** [review]

File: src/applications/disability-benefits/all-claims/content/contactInformation.jsx (hunk 4 above)
Comment (markgreenburg) [reply src/applications/disability-benefits/all-claims/content/contactInformation.jsx]: Yeah...I figured out how to make this better.

**Pull request merged**